posts/inside-rust/2023-08-21-clippy-perf-project.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: post
+title: "Making Clippy blazingly fast by 2024"
+author: Alejandra González
+team: the Clippy team <https://www.rust-lang.org/governance/teams/dev-tools#Clippy%20team>
+description: This is the start of a series of posts talking about optimizations done on the Clippy codebase. 
+---
+
+Clippy's codebase is designed to be beginner-friendly so new contributors can come and learn,
+not optimized for performance. It was never a priority to make Clippy fast because *it never needed
+to be.* But time passed and more lints were added, now we have [over 650 lints][lint_list], each one of them adding a bit of performance overhead. In big projects this can take over a minute to check all those lints, so the Clippy team have decided to do something about it.
```

Comment:
```suggestion
to be.* But time passed and more lints were added, now we have [over 650 lints][lint_list], each one of them adding a bit of performance overhead. In big projects this can take over a minute to check all those lints, so the Clippy team has decided to do something about it.
```